src/frequenz/sdk/_data_ingestion/battery_utils.py

```diff
@@ -0,0 +1,361 @@
+    ) -> None:
+        """Create class instance.
+
+        Args:
+            battery_id: Id of this battery
+            max_data_age_sec: If component stopped sending data, then
+                this is the maximum time when its last message should be considered as
+                valid. After that time, component won't be used until it starts sending
+                data.
+            max_blocking_duration_sec: This value tell what should be the maximum
+                timeout used for blocking failing component.
+        """
+        self._battery_id: int = battery_id
+        self._max_data_age: float = max_data_age_sec
+
+        self._init_method_called: bool = False
+        self._last_state: WorkingState = WorkingState.WORKING
+
+        self._min_blocking_duration_sec: float = 1.0
+        self._max_blocking_duration_sec: float = max_blocking_duration_sec
+        self._blocked_until: Optional[datetime] = None
+        self._last_blocking_duration_sec: float = self._min_blocking_duration_sec
+
+    @property
+    def battery_id(self) -> int:
+        """Get battery id.
+
+        Returns:
+            Battery id
+        """
+        return self._battery_id
+
+    async def init(self) -> None:
+        """Async part of constructor.
+
+        This should be called in order to subscribe for necessary data.
+
+        Raises:
+            RuntimeError: If given battery as no adjacent inverter.
+        """
+        component_graph = get_microgrid().component_graph
+        inverter_id = self._find_adjacent_inverter_id(component_graph)
+        if inverter_id is None:
+            raise RuntimeError(
+                f"Can't find inverter adjacent to battery: {self._battery_id}"
+            )
+
+        api_client = get_microgrid().api_client
+
+        bat_recv = await api_client.battery_data(self._battery_id)
+        # Defining battery_receiver and inverter receiver needs await.
+        # Because of that it is impossible to define it in constructor.
+        # Setting it to None first would require to check None condition in
+        # every call.
+        # pylint: disable=attribute-defined-outside-init
+        self._battery_receiver = _ComponentReceiver(
+            self._battery_id, bat_recv.into_peekable()
+        )
+        inv_recv = await api_client.inverter_data(inverter_id)
+        # pylint: disable=attribute-defined-outside-init
+        self._inverter_receiver = _ComponentReceiver(
+            inverter_id, inv_recv.into_peekable()
+        )
+
+        await asyncio.sleep(_START_DELAY_SEC)
+        self._init_method_called = True
+
+    def get_state(self) -> WorkingState:
+        """Return if battery message indicates that battery can be used.
+
+        Raises:
+            RuntimeError: if method `init` was not called or not awaited.
+
+        Returns:
+            True if battery can be used, False otherwise.
+        """
```

Comment:
The documentation seems to suggest that this returns a bool?

Reply:
Done